src/App.tsx

```diff
@@ -1,24 +1,192 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect, useMemo } from 'react';
+import {
+  getTodos,
+  createTodo,
+  deleteTodo,
+  updateTodo,
+} from './api/todos';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
+import { FilteredBy } from './types/FilteredBy';
+import { Todo } from './types/Todo';
+import { TodoNotification } from './components/TodoNotification';
 import { UserWarning } from './UserWarning';
 
-const USER_ID = 0;
+const USER_ID = 6662;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isTodoLoaded, setIsTodoLoaded] = useState(false);
+  const [filterBy, setFilterBy] = useState<FilteredBy>(FilteredBy.ALL);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isError, setIsError] = useState(false);
+
+  const isActiveTodos = useMemo(
+    () => todos.some(todo => todo.completed),
+    [todos],
+  );
+
+  const countActiveTodos = useMemo(
+    () => todos.filter(todo => !todo.completed).length,
+    [todos],
+  );
+
+  const getTodosFromServer = async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setIsTodoLoaded(true);
+      setTodos(todosFromServer);
+      setIsTodoLoaded(false);
+    } catch (error) {
+      setIsError(true);
+      setIsTodoLoaded(false);
+      setErrorMessage('Data couldn\'t be loaded from the server');
+    }
+  };
+
+  useEffect(() => {
+    getTodosFromServer();
+  }, []);
+
+  const createTodoOnServer = async (query: string) => {
+    const data = {
+      title: query,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    setTempTodo({
+      ...data,
+      id: 0,
+    });
+    try {
+      setIsTodoLoaded(true);
+      await createTodo(USER_ID, data);
+      await getTodosFromServer();
```

Comment:
would be better for performance to reduce requests on the server and update the array on the place. Same for all other manipulations with todos